x/datapool/types/pool.go

```diff
@@ -0,0 +1,72 @@
+package types
+
+import (
+	"fmt"
+	"strings"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/x/datadeal/v044_temp/address"
+)
+
+const (
+	PENDING = "PENDING"
+	ACTIVE  = "ACTIVE"
+)
+
+func NewPool(poolId uint64, curator sdk.AccAddress, poolParams PoolParams) *Pool {
+	poolAddress := NewPoolAddress(poolId)
+
+	return &Pool{
+		PoolId:       poolId,
+		PoolAddress:  poolAddress.String(),
+		Round:        1,
+		PoolParams:   &poolParams,
+		CurNumData:   0,
+		NumIssuedNft: 0,
+		Curator:      curator.String(),
+		Status:       PENDING,
+	}
+}
+
+func NewPoolAddress(poolId uint64) sdk.AccAddress {
+	key := append([]byte("pool"), sdk.Uint64ToBigEndian(poolId)...)
+	return address.Module(ModuleName, key)
+}
+
+func AccPoolAddressFromBech32(address string) (sdk.AccAddress, error) {
+	if len(strings.TrimSpace(address)) == 0 {
+		return sdk.AccAddress{}, fmt.Errorf("empty address string is not allowed")
+	}
+
+	bech32PrefixAccAddr := sdk.GetConfig().GetBech32AccountAddrPrefix()
+
+	bz, err := sdk.GetFromBech32(address, bech32PrefixAccAddr)
+	if err != nil {
+		return nil, err
+	}
+
+	err = VerifyPoolAddressFormat(bz)
+	if err != nil {
+		return nil, err
+	}
+
+	return sdk.AccAddress(bz), nil
+}
+
+func VerifyPoolAddressFormat(bz []byte) error {
+	verifier := sdk.GetConfig().GetAddressVerifier()
+	if verifier != nil {
+		return verifier(bz)
+	}
+
+	if len(bz) == 0 {
+		return sdkerrors.Wrap(sdkerrors.ErrUnknownAddress, "address cannot be empty")
+	}
+
+	if len(bz) > address.MaxAddrLen {
+		return sdkerrors.Wrapf(sdkerrors.ErrUnknownAddress, "address max length is %d, got %d", address.MaxAddrLen, len(bz))
+	}
```

Comment:
I think this part should be verified first.

Reply:
May I ask what does it mean? This verification is used for creating pool address. In that process, this verify the address is valid or not. What's the point I missed!?

Reply:
AHA. I mean when `bz` is `nil` when this logic is first executed, `nil` is entered when calling `verifier(bz)`, so it would be a problem.
However, if `nil` processing is done well in `verifier(bz)`, there semms to be no problem. 😀

Reply:
I got it thanks. I'll change it!

Reply:
modified in [5f69ad0](https://github.com/medibloc/panacea-core/pull/287/commits/5f69ad06762947b77e90421a4b2487f058772fa2)